src/test/scala/is/hail/expr/ir/MatrixIRSuite.scala

```diff
@@ -117,5 +118,30 @@ class MatrixIRSuite extends SparkSuite {
     val actualOrdering = getRows(unioned).map { case Row(i: Int) => i }
 
     assert(actualOrdering sameElements expectedOrdering)
+
+  }
+
+  @Test def testMatrixExplode() {
```

Comment:
I feel like we should generally be moving towards python tests. It'd also be easy there to assert that collecting the exploded field has the expected contents in the expected order.

Reply:
python tests already exist. I'm adding scala tests for this issue:
https://github.com/hail-is/hail/issues/4016

Reply:
What's the motivation for #4016 cc: @cseed?